services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java

```
@@ -241,16 +249,24 @@ public SignedUrl getSignedUrlWithCustomPolicy(Consumer<CustomSignerRequest.Build
      *                                                            .expirationDate(expirationDate)
      *                                                            .activeDate(activeDate)
      *                                                            .ipRange(ipRange)
+     *                                                            .resourceUrlPattern(resourceUrlPattern)
      *                                                            .build();
      *     SignedUrl signedUrl = utilities.getSignedUrlWithCustomPolicy(customRequest);
      *     String url = signedUrl.url();
      * }
      */
     public SignedUrl getSignedUrlWithCustomPolicy(CustomSignerRequest request) {
+        String resourceUrl = request.resourceUrl();
         try {
-            String resourceUrl = request.resourceUrl();
-            String policy = SigningUtils.buildCustomPolicyForSignedUrl(request.resourceUrl(), request.activeDate(),
-                                                                       request.expirationDate(), request.ipRange());
+            String resourceUrlPattern = StringUtils.isEmpty(request.resourceUrlPattern())
```

Is there any reason we are treating empty as null?

---

Because `resourceUrlPattern` maps to the JSON Policy document's `Resource` field, if a user provides `null` or `""` we default to the `resourceUrl` itself - meaning, there is no pattern, just use the base url as the `Resource` field in the policy. 

The server side implication of send a policy with a `Resource` field set to "" will result in a 403. I can't foresee a situation in which users are actively prohibiting all access to their distribution just by passing `""`.

For example: 

```java
        CustomSignerRequest request = CustomSignerRequest.builder()
                                                         .resourceUrl(resourceUrl)
                                                         .privateKey(keyFilePath)
                                                         .keyPairId(keyPairId)
                                                         .resourceUrlPattern("")
                                                         .activeDate(activeDate)
                                                         .expirationDate(expirationDate)
                                                         .build();
```
Will result in a 403.

---

Right, my question is more around: is this something we want to handle in the SDK? Should we just let it fail on the server side instead?

My concern is that some customers may not realize `""` is not supported or have bugs in their code to pass empty string to resourceUrlPattern unintentionally, which will be hidden by this SDK logic.

---

Ack. I don't feel strongly about it. I can just use a standard null check.